### PR TITLE
nc news #16

### PR DESCRIPTION
**Pull request by Clare30**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -3,6 +3,7 @@ const app = require("../app.js");
 const request = require("supertest");
 const testData = require("../db/data/test-data");
 const seed = require("../db/seeds/seed.js");
+const req = require("express/lib/request");
```
Comment (daviidbartlett) [inline __tests__/app.test.js:6]: what's this 👀

File: __tests__/app.test.js [hunk 2]
```diff
@@ -103,6 +104,64 @@ describe("/api/articles", () => {
...
+        });
+    });
+    test("status: 200 - filters by the topic entered in query", () => {
+      return request(app)
+        .get("/api/articles?topic=mitch")
+        .expect(200)
+        .then(({ body: { articles } }) => {
+          articles.forEach((article) => {
+            expect(article.topic).toBe("mitch");
+          });
```
Comment (daviidbartlett) [inline __tests__/app.test.js:138]: what about:
-  ?topic=banana
- ?topic=paper
I expect two different response from these

File: __tests__/app.test.js [hunk 3]
```diff
@@ -103,6 +104,64 @@ describe("/api/articles", () => {
...
+      return request(app)
+        .get("/api/articles?order=abcde")
+        .expect(400)
+        .then(({ body: { msg } }) => {
+          expect(msg).toBe("invalid query");
+        });
+    });
+    test("error: 400 - topic is not a valid topic query", () => {
+      return request(app)
+        .get("/api/articles?topic=fanta")
```
Comment (daviidbartlett) [inline __tests__/app.test.js:159]: 404 I reckon. Topic not found in topics table.
What about a topic that exists but has not articles? (paper)

**Review by Angaca — APPROVED** [review]
Looks good

**Pull request merged**